manifests/config.pp

```diff
@@ -165,6 +165,12 @@
           enable  => true,
           require => Package['sssd-dbus'],
         }
+
+        $ipa_config = '/etc/ipa/default.conf'
+        if find_file($ipa_config) {
+          $realm_entry = file($ipa_config).match('^realm\s*=\s*(.*)$')
```

Comment:
I don't think this does what you think it does. AFAIK `file()` runs on the Puppetserver so in a networked environment this will copy the realm from the Puppetserver to the clients. That's why we use facts for all the other entries